imap_processing/swe/swe_cdf_attrs.py

```diff
@@ -0,0 +1,217 @@
+            "at L1 to provide context for the ENA measurements and perform  "
+            "the in situ solar wind observations necessary to understand the"
+            "local structures that can affect acceleration and transport of "
+            "these particles. See"
+            "https://imap.princeton.edu/instruments/swe for more details."
+        )
+    ],
+    "Logical_file_id": ["FILL ME IN AT FILE CREATION"],
+} | cdf_utils.global_base
+
+
+# Required attrs for data type of
+# number, int and float.
+# We can create one attrs dict for both
+# with their default values.
+
+# For SWE, max range for uncompressed counts is
+# 65536. Therefore, I initialized it with that
+# value.
+int_attrs = {
+    "CATDESC": None,
+    "DISPLAY_TYPE": "spectrogram",
+    "FIELDNAM": None,
+    "FILLVAL": cdf_utils.INT_FILLVAL,
+    "FORMAT": "I18",
+    "LABLAXIS": None,
+    "UNITS": "int",
+    "VALIDMIN": np.int64(0),
+    "VALIDMAX": np.int64(65536),
+    "VAR_TYPE": "support_data",
+    "SCALETYP": "linear",
+}
+
+# For SWE, max range for uncompressed rates is
+# less than 800,000 when uncompressed count is
+# divided by 0.083 second.
+# TODO: check with SWE team if they want to keep
+# range in millieseconds or seconds and update this
+# max range accordingly.
+float_attrs = {
+    "CATDESC": None,
+    "DISPLAY_TYPE": "spectrogram",
+    "FIELDNAM": None,
+    "FILLVAL": cdf_utils.DOUBLE_FILLVAL,
+    "FORMAT": "F64.5",
+    "LABLAXIS": None,
+    "UNITS": "float",
+    "VALIDMIN": np.float64(-800000),
+    "VALIDMAX": np.float64(800000),
+    "VAR_TYPE": "support_data",
+    "SCALETYP": "linear",
+}
+
+# Housekeeping mode data array is stored as string.
+# Required attrs for string data type,
+# meaning array with string.
+string_attrs = {
+    "CATDESC": None,
+    "DEPEND_0": "Epoch",
+    "FORMAT": "A80",
+    "DISPLAY_TYPE": "no_plot",
+    "FIELDNAM": None,
+    "VAR_TYPE": "metadata",
+}
+
+
+# For each data level, create global attrs
+swe_l1a_global_attrs = {
+    "Data_type": "L1A>Level-1A",
+    "Logical_source": "imap_idex_l1a",
+    "Logical_source_description": "IMAP Mission SWE Instrument Level-1A Data.",
+} | swe_global_base
+
+swe_l1b_global_attrs = {
+    "Data_type": "L1B>Level-1B",
+    "Logical_source": "imap_idex_l1b",
+    "Logical_source_description": "IMAP Mission SWE Instrument Level-1B Data.",
+} | swe_global_base
+
+
+l1a_science_attrs = {
+    "CATDESC": "Science Data",
+    "DEPEND_0": "Epoch",
+    "DEPEND_1": "Energy",
+    "DEPEND_2": "Counts",
+    "DISPLAY_TYPE": "time_series",
+    "FIELDNAM": "Counts",
+    "FILLVAL": cdf_utils.INT_FILLVAL,
+    "FORMAT": "I18",
+    "LABLAXIS": "Counts",
+    "UNITS": "dN",
+    "VALIDMIN": np.int64(0),
+    "VALIDMAX": np.int64(65536),
+    "VAR_TYPE": "support_data",
+    "SCALETYP": "linear",
+}
+
+l1b_science_attrs = {
+    "CATDESC": "Science Data",
+    "DEPEND_0": "Epoch",
+    "DEPEND_1": "Energy",
+    "DEPEND_2": "Angle",
+    "DEPEND_3": "Rates",
```

Comment:
Not for you to worry about here, but an annoyance with CDF on my part... These DEPENDS seem like they should be automatically handled by specifying proper coordinates in xarray. My xarray datasets already know exactly what variables depend on which coordinates because it is a part of the spec, so that is already contained in the xarray metadata. This seems like we could abstract this out into the helper function to automatically add the DEPENDS based on the coordinates of the dataset... cc @maxinelasp

Reply:
It would be nice if we can abstract this out from xr metadata.